### PR TITLE
My Home: Remove old locations and cards

### LINKED ISSUE #41776: My Home: Remove old layouts
If https://github.com/Automattic/wp-calypso/pull/41712 lands and is stable, let's remember to remove old layouts and remove the experimental flag from code.

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

Removes the unused cards and locations after activating the new My Home layout. It also removes the experimental flag.

#### Testing instructions

- Go to My Home for a variety of sites (unlaunched, launched, established, WP for Teams).
- Make sure there are no changes when compared to production.

Fixes #41776

### CODE REVIEW

File: client/components/main/style.scss [hunk 1]
```diff
@@ -18,6 +18,14 @@
 	margin-top: 0;
 }
 
+@font-face {
```
Comment (mmtr) [inline client/components/main/style.scss:21]: This is kinda unrelated to this PR, but ensures the Recoleta font face is defined on all pages (we were previously setting in on My Home).

File: client/my-sites/customer-home/cards/actions/quick-links/index.jsx [hunk 2]
```diff
@@ -4,13 +4,10 @@
 import React from 'react';
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/actions/quick-links/index.jsx:4]: Note that these changes are actually caused by renaming `quick-links-compact` to `quick-links`.

File: client/my-sites/customer-home/cards/actions/quick-links/style.scss [hunk 3]
```diff
@@ -1,50 +1,29 @@
-.quick-links__boxes {
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/actions/quick-links/style.scss:1]: One more instance of changes caused by the rename of `quick-links-compact` to `quick-links`, feel free to ignore.

File: client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links/index.jsx [hunk 4]
```diff
@@ -4,13 +4,10 @@
 import React from 'react';
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/actions/wp-for-teams-quick-links/index.jsx:4]: Another rename here, so feel free to ignore (previous `wp-for-teams-quick-links` has been deleted, and `wp-for-teams-quick-links-compact` has been renamed to `wp-for-teams-quick-links`).

File: client/my-sites/customer-home/cards/features/stats/index.jsx [hunk 5]
```diff
@@ -1,136 +1,293 @@
 /**
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/features/stats/index.jsx:1]: Another rename here, so feel free to ignore (previous `stats` has been deleted, and `stats-v2` has been renamed to `stats`).

File: client/my-sites/customer-home/cards/features/stats/style.scss [hunk 6]
```diff
@@ -1,48 +1,111 @@
-.stats__data {
```
Comment (mmtr) [inline client/my-sites/customer-home/cards/features/stats/style.scss:1]: Another rename here, so feel free to ignore (previous `stats` has been deleted, and `stats-v2` has been renamed to `stats`).

**Comment by gwwar** [comment]
For next time, I think we could break this down into a couple of PRs to make it easier to review. We might miss something with changes mixed, but it's probably okay in this instance. So for example we might have decomposed this to three PRs
- Remove feature flag / component wiring
- Delete all unused components
- Move font definition

File: client/my-sites/customer-home/cards/actions/quick-links/index.jsx [hunk 7]
```diff
@@ -301,12 +299,15 @@ const mapDispatchToProps = {
...
 	addDomainAction,
+	expand: expandHomeQuickLinks,
+	collapse: collapseHomeQuickLinks,
 };
 
 const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
 	const { editHomePageUrl, isStaticHomePage, siteSlug } = stateProps;
 	return {
 		...stateProps,
+		...dispatchProps,
```
Comment (gwwar) [inline client/my-sites/customer-home/cards/actions/quick-links/index.jsx:310]: I think this was done on purpose originally to not include dispatchProps here. Did we need it?

Comment (gwwar) [reply client/my-sites/customer-home/cards/actions/quick-links/index.jsx:310]: Ah, I see pass through for expand/collapse. 👍

**Pull request merged**